core/src/main/java/com/linecorp/armeria/internal/common/stream/EmptyFixedStreamMessage.java

```diff
@@ -16,19 +16,51 @@
 
 package com.linecorp.armeria.internal.common.stream;
 
+import static java.util.Objects.requireNonNull;
+
 import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import org.reactivestreams.Subscriber;
 
 import com.google.common.collect.ImmutableList;
 
 import com.linecorp.armeria.common.annotation.Nullable;
 import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.common.stream.SubscriptionOption;
+import com.linecorp.armeria.common.util.UnmodifiableFuture;
+
+import io.netty.util.concurrent.EventExecutor;
 
 /**
  * A {@link FixedStreamMessage} that publishes no objects, just a close event.
  */
 @UnstableApi
 public class EmptyFixedStreamMessage<T> extends FixedStreamMessage<T> {
 
+    @Override
+    public boolean isComplete() {
+        return whenComplete().isDone();
+    }
+
+    @Override
+    public void subscribe(Subscriber<? super T> subscriber, EventExecutor executor,
+                          SubscriptionOption... options) {
+        requireNonNull(subscriber, "subscriber");
+        requireNonNull(executor, "executor");
+        executor.execute(() -> {
+            subscriber.onSubscribe(NoopSubscription.get());
+            subscriber.onComplete();
+            whenComplete().complete(null);
+        });
+    }
+
+    @Override
+    public CompletableFuture<List<T>> collect(EventExecutor executor, SubscriptionOption... options) {
+        whenComplete().complete(null);
+        return UnmodifiableFuture.completedFuture(ImmutableList.of());
```

Comment:
Cache the value?

Reply:
I didn't implement it because
- there's not so much chance that collect is calling more than once
- it's a light object which isn't any burden to make it twice.
If you still need that we need cache, please let me know. I will fix it. 😉

Reply:
>  I can at least add it to the UnmodifiableFuture.

I meant this. 😆